docs/guide/custom-functions.md

```diff
@@ -119,72 +108,243 @@ Register your function plugin (and its translations) so that HyperFormula can re
 
 Use the [`registerFunctionPlugin()`](../api/classes/hyperformula.md#registerfunctionplugin) method:
 
-```javascript
-HyperFormula.registerFunctionPlugin(CountHF, CountHF.translations);
+```js
+HyperFormula.registerFunctionPlugin(MyCustomPlugin, MyCustomPlugin.translations);
 ```
 
 ### 6. Use your custom function inside a formula
 
-Now, you can use your HYPER function inside a formula:
+Now, you can use your GREET function inside a formula:
 
-```javascript
+```js
 // prepare spreadsheet data
-const data = [['=HYPER()']];
+const data = [['Anthony', '=GREET(A1)']];
 
 // build a HyperFormula instance where you can use your function directly
 const hfInstance = HyperFormula.buildFromArray(data);
 
 // read the value of cell A1
-const result = hfInstance.getCellValue({ sheet: 0, col: 0, row: 0 });
+const result = hfInstance.getCellValue({ sheet: 0, col: 1, row: 0 });
 
 // open the browser's console to see the results
 console.log(result);
 ```
 
 ### Full example
 
-```javascript
-import { FunctionPlugin } from 'hyperformula';
+The complete implementation of this custom function is also included in the [demo](#working-demo).
 
-export class CountHF extends FunctionPlugin {
-  hyper(ast, state) {
-    return this.runFunction(ast.args, state, this.metadata('HYPER'), () => 'Hyperformula'.length);
+```js
+import { FunctionPlugin, CellError } from 'hyperformula';
+
+export class MyCustomPlugin extends FunctionPlugin {
+  greet(ast, state) {
+    return this.runFunction(
+      ast.args,
+      state,
+      this.metadata('GREET'),
+      (firstName) => {
+        return `👋 Hello, ${firstName}!`;
+      },
+    );
   }
 }
 
-CountHF.implementedFunctions = {
-  HYPER: {
-    method: 'hyper',
-  },
+MyCustomPlugin.implementedFunctions = {
```

Comment:
This line does not compile in TypeScript with the following error:

```
index.ts:23:16 - error TS2339: Property 'translations' does not exist on type 'typeof MyCustomPlugin'.

23 MyCustomPlugin.translations = {
```

The workaround is to add it as a filed in the class:

```
public static translations;
```

I think that you either need to refactor the snippet to define translations outside of this class, or add an optional field `translations` to `FunctionPlugin`

Reply:
In TS I would do:

```Typescript
export class MyCustomPlugin extends FunctionPlugin {
  static translations = {
    // ...
  }
```

but in plain JS it throws an error when run inside CodeSandbox:
![image](https://user-images.githubusercontent.com/2480086/208121884-1b080608-291a-4a7c-8fa2-43808a2a352d.png)

So I decided to keep it like this. Unless you know how to enable this "experimental syntax" in CodeSandbox ;)

Reply:
But do translations need to be provided as the static property of `MyCustomPlugin`? 

- If yes, then they should be added to the typings and my previous comment will be resolved.
- If not, then we will make these guides clearer, if we provide the translations as a separate object (e.g. `MyCustomPluginTranslations`).

Reply:
> But do translations need to be provided as the static property of MyCustomPlugin?

No, they could be a separate object. I get your point. You're right, that making it a member of the Plugin class overcomplicated the example. I'll change it.

Reply:
done